skbio/io/format/genbank.py

```diff
@@ -0,0 +1,865 @@
+as ``RNA``:
+
+>>> gb_rna = gb
+>>> gb_rna[-2] = gb_rna[-2].replace('t', 'u')
+>>> rna_seq_2 = RNA.read(gb_rna)
+>>> rna_seq_2 == rna_seq
+True
+
+>>> from io import StringIO
+>>> with StringIO() as fh:
+...     print(dna_seq.write(fh, format='genbank').getvalue())
+LOCUS       3K1V_A   34 bp   RNA   linear   SYN   10-OCT-2012
+DEFINITION  Chain A, Structure Of A Mutant Class-I Preq1.
+ACCESSION   3K1V_A
+VERSION     3K1V_A  GI:260656459
+KEYWORDS    .
+SOURCE      synthetic construct
+  ORGANISM  synthetic construct
+            other sequences; artificial sequences.
+REFERENCE   1  (bases 1 to 34)
+  AUTHORS   Klein,D.J., Edwards,T.E. and Ferre-D'Amare,A.R.
+  TITLE     Cocrystal structure of a class I preQ1 riboswitch
+  JOURNAL   Nat. Struct. Mol. Biol. 16 (3), 343-344 (2009)
+  PUBMED    19234468
+COMMENT     SEQRES.
+FEATURES             Location/Qualifiers
+       source        1..34
+                     /db_xref="taxon:32630"
+                     /mol_type="other RNA"
+                     /organism="synthetic construct"
+ORIGIN
+        1 agaggttcta gcacatccct ctataaaaaa ctaa
+//
+<BLANKLINE>
+
+References
+----------
+.. [1_] http://www.ncbi.nlm.nih.gov/nuccore/3K1V_A
+
+"""
+
+# ----------------------------------------------------------------------------
+# Copyright (c) 2013--, scikit-bio development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file COPYING.txt, distributed with this software.
+# ----------------------------------------------------------------------------
+
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+from future.builtins import range, zip
+
+import re
+import numpy as np
+import pandas as pd
+from datetime import datetime
+from functools import partial
+
+from skbio.io import create_format, GenBankFormatError
+from skbio.io.format._base import (
+    _get_nth_sequence, _line_generator, _too_many_blanks)
+from skbio.util._misc import chunk_str
+from skbio.sequence import Sequence, DNA, RNA, Protein
+
+
+genbank = create_format('genbank')
+
+# date format in locus line of genbank record
+_TIME_FORMAT = '%d-%b-%Y'
+# This list is ordered
+# used to read and write genbank file.
+_HEADERS = ['LOCUS',
+            'DEFINITION',
+            'ACCESSION',
+            'VERSION',
+            'DBSOURCE',
+            'DBLINK',
+            'KEYWORDS',
+            'SOURCE',
+            'REFERENCE',
+            'COMMENT',
+            'FEATURES',
+            'ORIGIN']
+
+
+@genbank.sniffer()
+def _genbank_sniffer(fh):
+    # check the 1st real line is a valid LOCUS line
+    if _too_many_blanks(fh, 5):
+        return False, {}
+    try:
+        line = next(_line_generator(fh, skip_blanks=True, strip=False))
+    except StopIteration:
+        return False, {}
+
+    try:
+        _parse_locus([line])
+    except GenBankFormatError:
```

Comment:
By header, you mean this thing

````
GBBCT1.SEQ          Genetic Sequence Data Bank
                          August 15 2015
 
                NCBI-GenBank Flat File Release 209.0
 
                     Bacterial Sequences (Part 1)
 
   63775 loci,    86300710 bases, from    63775 reported sequences
````

It will absolutely be stripped when it goes to NCBI.